spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskSchedulerController.java

```diff
@@ -95,9 +97,12 @@ public TaskSchedulerController(SchedulerService schedulerService) {
 	 */
 	@RequestMapping(value = "/{name}", method = RequestMethod.GET)
 	@ResponseStatus(HttpStatus.OK)
-	public ScheduleInfoResource getSchedule(
-			@PathVariable("name") String scheduleName) {
-		return taskAssembler.toResource(this.schedulerService.getSchedule(scheduleName));
+	public ScheduleInfoResource getSchedule(@PathVariable("name") String scheduleName) {
+		ScheduleInfo scheduler = this.schedulerService.getSchedule(scheduleName);
+		if (scheduler == null) {
+			throw new NoSuchTaskSchedulerException(String.format("Task scheduler [%s] doesn't exist!" , scheduleName));
```

Comment:
Given this is actually getting the `schedule` information not the `scheduler` information, it would be better to use `schedule` instead fo `scheduler`. Also, we may want to call this exception as `NoSuchScheduleException` as `NoSuchTaskSchedulerException` may not represent the actual exception being thrown.